src/she/sdl2/sdl2_display.cpp

```diff
@@ -0,0 +1,363 @@
+  }
+
+  int SDL2Display::originalHeight() const
+  {
+    return m_restoredHeight > 0 ? m_restoredHeight: height();
+  }
+
+  void SDL2Display::setOriginalWidth(int width)
+  {
+    std::cout << "New Original width: " << std::to_string(width) << std::endl;
+    m_restoredWidth = width;
+  }
+
+  void SDL2Display::setOriginalHeight(int height)
+  {
+    std::cout << "New Original height: " << std::to_string(height) << std::endl;
+    m_restoredHeight = height;
+  }
+
+  int SDL2Display::scale() const
+  {
+    return m_scale;
+  }
+
+  void SDL2Display::setScale(int scale)
+  {
+    ASSERT(scale >= 1);
+    if (m_scale == scale)
+      return;
+
+    m_scale = scale;
+    recreateSurface();
+  }
+
+  void SDL2Display::recreateSurface()
+  {
+    auto  newSurface = new SDL2Surface(width() / m_scale, height() / m_scale, SDL2Surface::DeleteAndDestroy);
+    if (m_surface) {
+      m_surface->blitTo(newSurface, 0, 0, 0, 0, width(), height());
+      m_surface->dispose();
+    }
+    m_dirty = true;
+    m_surface = newSurface;
+    she::sdl::screen = newSurface;
+  }
+
+  Surface* SDL2Display::getSurface()
+  {
+    return m_surface;
+  }
+
+  void SDL2Display::present() {
+    if (!m_dirty)
+      return;
+    m_dirty = false;
+    SDL_UpdateWindowSurface(m_window);
+  }
+
+  void SDL2Display::flip(const gfx::Rect& bounds)
+  {
+    m_dirty = true;
+
+    auto nativeSurface = SDL_GetWindowSurface(m_window);
+    SDL_Rect rect {bounds.x, bounds.y, bounds.w, bounds.h};
+    SDL_Rect dst  {bounds.x * m_scale, bounds.y * m_scale, bounds.w * m_scale, bounds.h * m_scale};
+    SDL_BlitScaled((SDL_Surface*)m_surface->nativeHandle(), &rect, nativeSurface, &dst);
+  }
+
+  void SDL2Display::maximize()
+  {
+    SDL_MinimizeWindow(m_window);
+  }
+
+  bool SDL2Display::isMaximized() const
+  {
+    return sdl::isMaximized;
+  }
+
+  bool SDL2Display::isMinimized() const
+  {
+    return sdl::isMinimized;
+  }
+
+  void SDL2Display::setTitleBar(const std::string& title)
+  {
+    SDL_SetWindowTitle(m_window, title.c_str());
+  }
+
+  NativeCursor SDL2Display::nativeMouseCursor()
+  {
+    return m_nativeCursor;
+  }
+
+  bool SDL2Display::setNativeMouseCursor(NativeCursor cursor)
+  {
+    return false;
+
+    if (cursor == m_nativeCursor) {
+      return true;
+    }
+
+    if (m_cursor)
+      SDL_FreeCursor(m_cursor);
+    m_cursor = SDL_DISABLE;
+
+    switch (cursor) {
+    case kNoCursor: break;
+    case kArrowCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_ARROW); break;
+    case kIBeamCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_IBEAM); break;
+    case kWaitCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_WAIT); break;
+    // case kHelpCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HELP); break;
+    case kForbiddenCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_NO); break;
+    case kMoveCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HAND); break;
```

Comment:
This should probably be SDL_SYSTEM_CURSOR_SIZEALL

Reply:
With the amount of cursors that are unavailable, it's probably better to simply remove native cursor support entirely.

Reply:
Alternatively, we could stick to the available subset.

Reply:
It would be strange having some of the directions but not others.

Reply:
We could probably map kSizeNCursor to SYSTEM_CURSOR_SIZENS, kSizeNECursor to SYSTEM_CURSOR_SIZENESW, etc. While they aren't exactly the same, they do both mean somewhat the same thing.